### PR TITLE
Mark EDX as trashed by the no-gc helper calls

### LINKED ISSUE #4666: RyuJIT x86: Silent bad codegen in Linq.cmd
JIT\Performance\CodeQuality\Linq\Linq\Linq.cmd

test output:

```
Expected: 100
Actual: -1
```

### PR DESCRIPTION
**Pull request by BruceForstall**
Fixes #4666

@CarolEidt @adiaaida PTAL
/cc @dotnet/jit-contrib

### CODE REVIEW

File: src/jit/target.h [hunk 1]
```diff
@@ -558,7 +558,7 @@ typedef unsigned short          regPairNoSmall; // arm: need 12 bits
   // We don't allow using ebp as a source register. Maybe we should only prevent this for ETW_EBP_FRAMED (but that is always set right now).
   #define RBM_WRITE_BARRIER_SRC    (RBM_EAX|RBM_ECX|RBM_EBX|RBM_ESI|RBM_EDI)
 
-  #define RBM_CALLEE_TRASH_NOGC    RBM_NONE
+  #define RBM_CALLEE_TRASH_NOGC    RBM_EDX
```
Comment (sivarv) [inline src/jit/target.h]: Is this a bug on Legacy x86 JIT as well?

Comment (BruceForstall) [inline src/jit/target.h]: No, because the legacy codegen does trash the register using the old-style register allocation.

**Comment by sivarv** [comment]
Looks good.

**Comment by adiaaida** [comment]
LGTM

**Comment by briansull** [comment]
LGTM

**Comment by BruceForstall** [comment]
There's always some new fun with the Ubuntu/OSX legs :-(

Ubuntu failure:
```
Directory specified by --testNativeBinDir does not exist: /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_tst_prtest/bin/obj/Linux.x64.Checked/tests/src
```

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by BruceForstall** [comment]
@mmitche Some weird Ubuntu failures with this PR:
```
18:54:31  GitHub pull request #4810 of commit f6796220664d3bfa4ea42d2ab86cf5a6e2edeae9, no merge conflicts.
18:54:31 ln builds/lastSuccessfulBuild /jenkins/jobs/dotnet_coreclr/jobs/master/jobs/checked_ubuntu_tst_prtest/lastSuccessful failed
18:54:31 java.nio.file.FileAlreadyExistsException: /jenkins/jobs/dotnet_coreclr/jobs/master/jobs/checked_ubuntu_tst_prtest/lastSuccessful
18:54:31 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88)
18:54:31 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
18:54:31 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
18:54:31 	at sun.nio.fs.UnixFileSystemProvider.createSymbolicLink(UnixFileSystemProvider.java:456)
18:54:31 	at java.nio.file.Files.createSymbolicLink(Files.java:996)
18:54:31 	at sun.reflect.GeneratedMethodAccessor677.invoke(Unknown Source)
18:54:31 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:54:31 	at java.lang.reflect.Method.invoke(Method.java:606)
18:54:31 	at hudson.Util.createSymlinkJava7(Util.java:1235)
18:54:31 	at hudson.Util.createSymlink(Util.java:1151)
18:54:31 	at hudson.model.Run.createSymlink(Run.java:1840)
18:54:31 	at hudson.model.Run.updateSymlinks(Run.java:1821)
18:54:31 	at hudson.model.Run.execute(Run.java:1736)
18:54:31 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
18:54:31 	at hudson.model.ResourceController.execute(ResourceController.java:98)
18:54:31 	at hudson.model.Executor.run(Executor.java:410)
```
and:
```
18:56:24 Directory specified by --testNativeBinDir does not exist: /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_tst_prtest/bin/obj/Linux.x64.Checked/tests/src
18:56:24 Build step 'Execute shell' marked build as failure
```

**Comment by BruceForstall** [comment]
OSX also has:
```
19:18:00 Directory specified by --testNativeBinDir does not exist: /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx_tst_prtest/bin/obj/OSX.x64.Checked/tests/src
19:18:01 Build step 'Execute shell' marked build as failure
```

**Pull request merged**